docs/decisions/0018-units-as-containers.rst

```diff
@@ -0,0 +1,78 @@
+
+Decisions
+---------
+
+All decisions from `0017-generalized-containers.rst`_ are still valid but are written here alongside unit-specific decisions for better illustration.
+
+.. _`0017-generalized-containers.rst`: 0017-generalized-containers.rst
+
+1. Units as Containers
+=======================
+
+- A unit is a concrete type of container that holds components.
+- A unit is a container, making it also a publishable entity.
+- A Django application, which builds on the container application definitions, will an API and enough definitions for other unit subtypes to use.
+
+2. Unit Types and Content Constraints
+======================================
+
+- Units can only hold components as their members but will not enforce this restriction at the model level.
+- Units are the first level of nested content types Unit > Components.
+- Content restrictions for units are implemented at the app layer, allowing units to limit their members to only components.
+- Unit subtypes can be created by following the extensibility principles in `0003-content-extensibility.rst`_.
+
+3. Unit Members and Relationships
+==================================
+
+- The members of a unit can only be components.
+- Components are referenced as an ordered list in a unit.
+- Units can hold both static and dynamic content, such as user-specific variations.
+- Units can reference pinned and unpinned versions of its components.
+- The latest draft or publish version of a component can be set by using `None` in thr parent-child relationship between units-components.
+- A single component can be reference by multiple units.
+
+4. Unit Version History
+============================
+
+- Each unit version holds different list of components to support rollback operations and history tracking.
+- The author-defined list is the list of components defined by the author for a specific unit version.
+- The author-defined list of components won't change for a specific version.
+- The initial list is a copy of the author-defined list that has all components pinned to the versions at the time of the unit version creation.
+- The initial list is immutable for a unit version.
+- The frozen list refers to the list of components at the time when the next version of the unit is created.
+- When creating the author-defined list of a new version with pinned references, then the author-defined list is the same as the initial and frozen list. When creating a new version with unpinned references, then the frozen list starts as `None` and should be updated with the author-defined components pinned when a new version is created.
+- The author-defined list is used to show the content of a unit version as the author specified it, the frozen list can be used for discard operations on a draft version and the initial-list is part of the history of evolution of the unit.
+- These lists allow history tracking of a unit version and revert operations.
```

Comment:
note to self: This needs to be updated according the latest changes in the container ADR.